### ISSUE TITLE
Make terminal accessibility buffer read only

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#172006]
Terminal a11y buffer is not read-only.
Since this is contenteditable textbox, unintended delete key can mess up the output content. I confirmed that delete key works inside the buffer.

It would be even nicer if this buffer could be marked up with readonly to prevent users from accidentally deleting their output content.

https://github.com/microsoft/vscode/issues/171914#issue-1551817935

**Comment by meganrogge** [comment]
@jooyoungseo, did you notice that when you toggle the mode it comes back as the original? so while it seems like editing of the buffer is possible, it's not actually making any changes.

**Comment by jooyoungseo** [comment]
@meganrogge -- You are correct. If you press tab and come back to the buffer via Shift+Tab, it comes back to the original content. However, the content is editable when it gets focused.

**Comment by meganrogge** [comment]
Not sure if it's possible to change this, but will investigate

**Comment by meganrogge** [comment]
Looks like this is easy to do - just disabling `contentEditable` and setting `readOnly`

**Comment by jooyoungseo** [comment]
@meganrogge -- To give you prompt testing results, I have just built the latest main branch on my end (this issue will also be found tomorrow once insider is released). I have found the following inadvertent regression (tested with NVDA and JAWS on Windows):

1. terminal a11y buffer is not readable via standard arrow keys.

1. Terminal input -> terminal a11y buffer: only requires one Shift+Tab. However, from terminal a11y back to terminal input requires two tab key press.

Note that aria-live is now working!

**Comment by meganrogge** [comment]
@jooyoungseo I have already caught that regression, you can build based on this PR to see that I've fixed it 😄 https://github.com/microsoft/vscode/pull/172276

**Comment by meganrogge** [comment]
re #2, i'll investigate that

**Comment by pawin35** [comment]
@meganrogge  As of today's insider build (Commit: 07d6f5b35fac559504dec65e3073d796b2d25a55), it appears that the accessibility buffer can still be modified accidentally. Additionally, if the buffer has been altered, tabbing to the shell and then shift-tabbing back does not restore the buffer to its previous state, as it did in previous versions. Furthermore, if the buffer has been modified, subsequent command output will not be reflected in the buffer.

**Comment by meganrogge** [comment]
> it appears that the accessibility buffer can still be modified accidentally

@pawin35 see https://github.com/microsoft/vscode/pull/172276#issuecomment-1402821577 for why we didn't go with readonly. the actual buffer is never modified - the text area element is. focusing the terminal buffer via mouse click for example and shift+tabbing again will restore the contents to what they were before.

**Comment by pawin35** [comment]
@meganrogge  Is there a way to reset the state of the buffer without using the mouse? As of the version on the main branch, I am unable to reset the buffer by tabbing and shift-tabbing back to it.

**Comment by meganrogge** [comment]
Just hitting tab works for me @pawin35

**Comment by pawin35** [comment]
@jooyoungseo  Can you get the buffer to reset after tabbing into it after modifying the buffer?
I cannot get the buffer to reset with the NVDA on Windows 11; not sure weather there is something wrong on my ends.

**Comment by jooyoungseo** [comment]
@pawin35 -- I can reproduce your issue. Please file a separate issue and provide minimal reproducible steps so that @meganrogge can fully replicate this issue.

**Comment by pawin35** [comment]
@meganrogge  @jooyoungseo  I have created the issue at 172714.
